terraform/resource_provider_mock.go

```diff
@@ -56,6 +56,8 @@ type MockResourceProvider struct {
 	ReadDataDiffFn                 func(*InstanceInfo, *ResourceConfig) (*InstanceDiff, error)
 	ReadDataDiffReturn             *InstanceDiff
 	ReadDataDiffReturnError        error
+	TestResetCalled                bool
+	TestResetError                 error
```

Comment:
Seems messy to add this interface here since the core doesn't care about it all. Could we modify the test in helper/resource to add it to a custom provider to verify it is called?